make/hotspot/gensrc/GensrcAdlc.gmk

```diff
@@ -133,6 +133,18 @@ ifeq ($(call check-jvm-feature, compiler2), true)
     ADLCFLAGS += -DARM=1
   endif
 
+  # Set ASSERT, NDEBUG and PRODUCT flags just like in JvmFlags.gmk
+  ifeq ($(DEBUG_LEVEL), release)
+    ADLCFLAGS += -DNDEBUG
```

Comment:
May be you should also copy all comments from `JvmFlags.gmk` to avoid confusion because, for example, `NDEBUG` is not used directly in HotSpot code but it is needed "to disable uses of assert macro from <assert.h>."

Reply:
Done